### PR TITLE
Add Flite to amp-ads feature

### PR DESCRIPTION
**Pull request by andyhills**
Adding Flite to amp-ads per issue [#1983](https://github.com/ampproject/amphtml/issues/1983)

Google CLA is under company Flite.

### CODE REVIEW

File: ads/flite.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ export function flite(global, data) {
+   checkData(data, ['guid','mixins']);
+   const guid = data.guid, o = global, e = encodeURIComponent, x = 0;
+   let r = "", m, url, dep = "";
+   o.FLITE = o.FLITE || {};
+   o.FLITE.config = o.FLITE.config || {};
+   o.FLITE.config[guid] = o.FLITE.config[guid] || {};
+   o.FLITE.config[guid].cb = Math.random();
+   o.FLITE.config[guid].ts = (+Number(new Date()));
+   r = global.context.location.href;
```
Comment (cramforce) [inline ads/flite.js]: Just wanted to understand whether you want to use this, which can be a URL like "https://cdn.ampproject.org/c/example.com/path" or whether you'd rather use `window.context.canonicalUrl` which points to the publisher's origin document.

File: ads/flite.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+
+ export function flite(global, data) {
+   checkData(data, ['guid','mixins']);
+   const guid = data.guid, o = global, e = encodeURIComponent, x = 0;
+   let r = "", m, url, dep = "";
```
Comment (cramforce) [inline ads/flite.js]: Single quoted strings please. I'm very surprised our linter is not unhappy

File: ads/flite.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+
+ import {loadScript, checkData} from '../src/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+
```
Comment (cramforce) [inline ads/flite.js]: Remove newline

**Comment by cramforce** [comment]
A few nits and a question. Looks great overall.

File: ads/flite.js (hunk 1 above)
Comment (andyhills) [inline ads/flite.js]: We are happy to know this so we can differentiate as to where the ad is actually serving. We are interested in knowing both about the origin page and the cached version.

Comment (cramforce) [inline ads/flite.js]: This is fine, of course. Just wanted to make sure you made a conscious decision.

**Comment by andyhills** [comment]
@cramforce can you elaborate on what would be causing the test 'gulp test --compiled' to fail?

**Comment by cramforce** [comment]
I think that was a flake. I restarted the test.

**Comment by andyhills** [comment]
Thank you.

**Pull request merged**